Zend/zend_attributes_arginfo.h

```diff
@@ -87,59 +87,50 @@ static zend_class_entry *register_class_Attribute(void)
 	ZVAL_LONG(&const_TARGET_CLASS_value, ZEND_ATTRIBUTE_TARGET_CLASS);
 	zend_string *const_TARGET_CLASS_name = zend_string_init_interned("TARGET_CLASS", sizeof("TARGET_CLASS") - 1, 1);
 	zend_declare_typed_class_constant(class_entry, const_TARGET_CLASS_name, &const_TARGET_CLASS_value, ZEND_ACC_PUBLIC, NULL, (zend_type) ZEND_TYPE_INIT_MASK(MAY_BE_LONG));
-	zend_string_release(const_TARGET_CLASS_name);
```

Comment:
I would think about specialized functions that accept ``char*`` and ``size_t```.
Also may be add specialization by type.
Actually, we already have ``zend_declare_class_constant_long()`` that we used before...
Probably API should be adjusted according to the new requirements.

Reply:
Slightly surprising, the difference is that `zend_declare_class_constant_long` (and `zend_declare_class_constant` etc) create a constant whose value is a long but has no type declaration:

https://github.com/php/php-src/blob/9283abc1a0f10c15258ddcafa2b4d72b2c6e3237/Zend/zend_API.c#L4910